src/urllib3/connection.py

```diff
@@ -230,22 +309,32 @@ def request_chunked(self, method, url, body=None, headers=None):
 
         if body is not None:
             if isinstance(body, (str, bytes)):
-                body = (body,)
-            for chunk in body:
-                if not chunk:
-                    continue
-                if not isinstance(chunk, bytes):
-                    chunk = chunk.encode("utf8")
-                len_str = hex(len(chunk))[2:]
-                to_send = bytearray(len_str.encode())
-                to_send += b"\r\n"
-                to_send += chunk
-                to_send += b"\r\n"
-                self.send(to_send)
+                self.send_chunk(body)
+            else:
+                for chunk in body:
+                    self.send_chunk(chunk)
```

Comment:
Is this refactor necessary? Typically with type hint additions we don't make code changes if possible.

Reply:
I think it is necessary here, because in the `if isinstance(body, (str, bytes))` branch mypy knows `(body,)` is `Tuple[Union[bytes, str]]` and doesn't allow the `body = (body,)` assignment as http body is `HTTPBody = Union[bytes, IO[Any], Iterable[bytes], str]`.

Reply:
Can `HTTPBody` be expanded to also have `Iterable[str]` and then does the previous implementation work?

Reply:
I changed the type of `request_chunked`'s `body` parameter to be `Union[None, HTTPBody, Tuple[Union[bytes, str]]` and brought back original implementation.